### PR TITLE
SmartCardContext internal slots and listReaders() algo

### PR DESCRIPTION
**Pull request by dandrader**
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/web-smart-card/pull/13.html" title="Last updated on Jul 21, 2023, 9:51 AM UTC (8368ed6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/web-smart-card/13/12c28f2...8368ed6.html" title="Last updated on Jul 21, 2023, 9:51 AM UTC (8368ed6)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
...
+      </table>
       <section>
         <h3><dfn>listReaders()</dfn> method</h3>
-        <div class="issue">Write an algorithm for this method.</div>
+        <p>The {{SmartCardContext/listReaders()}} method steps
+        are:</p>
+        <ol>
+          <li>If [=this=].{{SmartCardContext/[[operationInProgress]]}} is
+            `true`, [=exception/throw=] a "{{InvalidStateError}}"
+            {{DOMException}}.</li>
```
Comment (reillyeon) [inline index.html:187]: In other specifications I've tried to be consistent about either using exceptions or rejecting a promise but not both. I would rewrite this as:

```suggestion
          <li>Let |promise:Promise| be [=a new promise=].</li>
          <li>If [=this=].{{SmartCardContext/[[operationInProgress]]}} is
            `true`, [=reject=] |promise| with a "{{InvalidStateError}}"
            {{DOMException}} and return |promise|.</li>
```

File: index.html [hunk 2]
```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
...
+          <li>Let |promise:Promise| be [=a new promise=].</li>
+          <li>Run the following steps [=in parallel=]:
+            <ol>
+              <li>Let |resourceQuery:RESOURCEQUERY| be a new instance of the
+                platform's [[PCSC5]] `RESOURCEQUERY` class, with
+                [=this=].{{SmartCardContext/[[resourceManager]]}} as its
+                contructor input parameter.</li>
+              <! –– NB: Can't have "[]" in the data type part of a variable declaration. ––>
+              <li>Let |groups:array of STR| be the platform's [[PCSC5]] `STR[]`
+                containing the list of groups names that is equivalent to "all
```
Comment (reillyeon) [inline index.html:199]:
```suggestion
                containing the list of group names that is equivalent to "all
```

File: index.html [hunk 3]
```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
...
+                readers in the system" in that platform.
+                <aside class="note">This could also be an empty `STR[]` in some
```
Comment (reillyeon) [inline index.html:201]:
```suggestion
                <aside class="note">This could be an empty `STR[]` in some
```

File: index.html [hunk 4]
```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
...
+              <li>Otherwise, [=queue a global task=] on the [=relevant global
+                object=] of [=this=] using the [=smart card task source=] which
+                performs the following steps:
+                <ol>
+                  <li>Set [=this=].{{SmartCardContext/[[operationInProgress]]}} to
+                    `false`.</li>
+                  <li>[=resolve=] |promise| with a `sequence` of {{DOMString}}
+                    equivalent to |pcscReaders|.</li>
+                </ol>
+              </li>
```
Comment (reillyeon) [inline index.html:230]: Refactor this since you always queue a global tasks and set `operationInProgress` to `false`.

File: index.html [hunk 5]
```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
...
+                    {{SmartCardError/corresponding}} to |responseCode|.</li>
+                </ol>
+              </li>
+              <li>Otherwise, [=queue a global task=] on the [=relevant global
+                object=] of [=this=] using the [=smart card task source=] which
+                performs the following steps:
+                <ol>
+                  <li>Set [=this=].{{SmartCardContext/[[operationInProgress]]}} to
+                    `false`.</li>
+                  <li>[=resolve=] |promise| with a `sequence` of {{DOMString}}
```
Comment (reillyeon) [inline index.html:227]:
```suggestion
                  <li>[=Resolve=] |promise| with a `sequence` of {{DOMString}}
```

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (dandrader) [reply index.html]: Done.

File: index.html (hunk 4 above)
Comment (dandrader) [reply index.html]: Done.

**Review by reillyeon — APPROVED** [review]

**Pull request merged**